### PR TITLE
colima: use lima-bin on darwin for native macOS virtualization support

### PR DESCRIPTION
**Pull request by tricktron**
###### Description of changes

Tested with `result/bin/colima start --vm-type vz`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pecigonzalo** [comment]
<3

File: pkgs/applications/virtualization/colima/default.nix [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 , installShellFiles
 , lima
+, lima-bin
 , makeWrapper
 , qemu
 , testers
 , colima
 }:
 
+let lima-drv = if stdenv.isDarwin then lima-bin else lima;
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/colima/default.nix:14]: We should add a really short comment why we want this.

Comment (tricktron) [reply pkgs/applications/virtualization/colima/default.nix:16]: @SuperSandro2000 Good idea. I added a comment and a link to the pull request for more infos.

File: pkgs/applications/virtualization/colima/default.nix [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 , makeWrapper
 , qemu
 , testers
 , colima
 }:
 
+# use lima-bin on darwin to support native macOS virtualization
+# https://github.com/NixOS/nixpkgs/pull/209171
+let lima-drv = if stdenv.isDarwin then lima-bin else lima;
+in
```
Comment (wegank) [inline pkgs/applications/virtualization/colima/default.nix:17]:
```suggestion
, colima
  # use lima-bin on darwin to support native macOS virtualization
  # https://github.com/NixOS/nixpkgs/pull/209171
, lima-drv ? if stdenv.isDarwin then lima-bin else lima
}:

```

I would like this to be configurable, in my opinion.

Comment (tricktron) [reply pkgs/applications/virtualization/colima/default.nix]: Great suggestion. Thanks!

**Review by wegank — APPROVED** [review]

File: pkgs/applications/virtualization/colima/default.nix [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 , installShellFiles
 , lima
+, lima-bin
 , makeWrapper
 , qemu
 , testers
 , colima
+  # use lima-bin on darwin to support native macOS virtualization
+  # https://github.com/NixOS/nixpkgs/pull/209171
+, lima-drv ? if stdenv.isDarwin then lima-bin else lima
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/colima/default.nix:14]: nit I would probably inline this unless it is desirable to be overwritable

**Pull request merged**